### ISSUE TITLE
I am little bit confused. Isn't this training code block similar to backpropogation?

### ISSUE THREAD

**Issue by nikilpatel94** [ashishbamania/Tutorials-On-Artificial-Intelligence#1]
https://github.com/ashishbamania/Tutorials-On-Artificial-Intelligence/blame/ab13d6b6b349aeac5f342b66cf924fdb5396011f/Training Without Backpropagation/NoPropDT_on_MNIST.ipynb#L231-L234

Within the `train_nopropdt()`

```
optimizer.zero_grad()
loss.backward()
torch.nn.utils.clip_grad_norm_(model.parameters(), 1.0)
optimizer.step()
```

**Comment by ashishbamania** [comment]
Yes, but the gradients are not shared or backpropagated between all layers.
